api/enki.py

```diff
@@ -233,11 +249,40 @@ def reaper_request(self, identifier):
 
             return circulationdata
 
-    def checkout(self, patron, pin, licensepool, internal_format):
-        # WIP.
-        return None
+    def epoch_to_struct(self, epoch_string):
+        # This will turn the time string we get from Enki into a
+        # struct that the Circulation Manager can make use of.
+        time_format = "%Y-%m-%dT%H:%M:%S"
+        return datetime.datetime.strptime(
+            time.strftime(time_format, time.gmtime(float(epoch_string))),
+            time_format
+        )
 
-        # Create the loan info. We don't know the expiration 
+
+    def checkout(self, patron, pin, licensepool, internal_format):
+        identifier = licensepool.identifier
+        enki_id = identifier.identifier
+        response = self.loan_request(patron.authorization_identifier, pin, enki_id)
+        if response.status_code != 200:
+            raise CannotLoan(response.status_code)
+        result = json.loads(response.content)['result']
+        if not result['success']:
+            message = result['message']
+            if "That record is already checked out to you" in message:
+                # this might be okay
+                pass
+            elif "There are no available copies" in message:
+                # TODO: raise no copies error
+                self.log.error("There are no copies of this book available.")
+                raise CirculationException()
+            elif "Login unsuccessful" in message:
+                # TODO: raise invalid barcode/password error
+                self.log.error("User validation against Enki server was unsuccessful.")
+                raise CirculationException()
```

Comment:
Raise AuthorizationFailedException() here. All these exceptions are in circulation_exceptions.py.